src/js/components/todoList.js

```diff
@@ -0,0 +1,114 @@
+import TodoItem from './todoItem.js';
+import { ALL, VIEW, COMPLETE, USER_API, NONE, convertToPriority } from '../constant/constant.js';
+import { validLength } from '../utils/utils.js';
+
+class TodoList {
+  constructor(
+    $target,
+    { filter, onDeleteItem, changeTodoState, changeTodoValue, changeTodoPriority, dataLoader }
+  ) {
+    this.$target = $target;
+    this.state = {
+      user: {
+        name: '',
+        _id: '',
+        todoList: []
+      },
+      filter
+    };
+    this.dataLoader = dataLoader;
+    this.render();
+    this.addEvent(onDeleteItem, changeTodoState, changeTodoValue, changeTodoPriority);
+  }
+
+  setState = (nextState) => {
+    this.state = nextState;
+    this.render();
+  };
+
+  addEvent = (onDeleteItem, changeTodoState, changeTodoValue, changeTodoPriority) => {
+    this.$target.addEventListener('click', async ({ target, target: { className }}) => {
+      const closestLi = target.closest('li');
+      const index = closestLi.dataset['index'];
+      const id = this.state.user._id;
+      const itemId = this.state.user.todoList[+index]._id;
```

Comment:
이렇게 빈번하게 쓰이는 내용을 함수로 변경하는게 좋을 것 같다.
```suggestion
      const { closestLi, index, id, itemId } = getInfo();
```